micrometer-core/src/main/java/io/micrometer/core/instrument/binder/http/AbstractDefaultHttpClientRequestObservationConvention.java

```diff
@@ -0,0 +1,140 @@
+import io.micrometer.core.instrument.binder.http.HttpObservationDocumentation.CommonHighCardinalityKeys;
+import io.micrometer.core.instrument.binder.http.HttpObservationDocumentation.CommonLowCardinalityKeys;
+
+/**
+ * Abstract convention methods to share between javax and jakarta based implementations.
+ *
+ * For internal use, do not inherit.
+ *
+ * @author Marcin Grzejsczak
+ * @since 1.12.0
+ */
+@Incubating(since = "1.12.0")
+public class AbstractDefaultHttpClientRequestObservationConvention {
+
+    private static final Pattern PATTERN_BEFORE_PATH = Pattern.compile("^https?://[^/]+/");
+
+    private static final KeyValue URI_NONE = KeyValue.of(CommonLowCardinalityKeys.URI, KeyValue.NONE_VALUE);
+
+    private static final KeyValue METHOD_NONE = KeyValue.of(CommonLowCardinalityKeys.METHOD, KeyValue.NONE_VALUE);
+
+    private static final KeyValue STATUS_CLIENT_ERROR = KeyValue.of(CommonLowCardinalityKeys.STATUS, "CLIENT_ERROR");
+
+    private static final KeyValue CLIENT_NAME_NONE = KeyValue.of(ClientLowCardinalityKeys.CLIENT_NAME,
+            KeyValue.NONE_VALUE);
+
+    private static final KeyValue EXCEPTION_NONE = KeyValue.of(CommonLowCardinalityKeys.EXCEPTION, KeyValue.NONE_VALUE);
+
+    private static final KeyValue HTTP_URL_NONE = KeyValue.of(CommonHighCardinalityKeys.HTTP_URL, KeyValue.NONE_VALUE);
+
+    private static final KeyValue USER_AGENT_NONE = KeyValue.of(CommonHighCardinalityKeys.USER_AGENT_ORIGINAL,
+            KeyValue.NONE_VALUE);
+
+    protected String getContextualName(String lowercaseMethod) {
+        return "http " + lowercaseMethod;
+    }
+
+    protected KeyValues getLowCardinalityKeyValues(@Nullable URI uri, @Nullable Throwable throwable,
+            @Nullable String methodName, @Nullable Integer statusCode, @Nullable String uriPathPattern) {
+        return KeyValues.of(clientName(uri), exception(throwable), method(methodName), outcome(statusCode),
+                status(statusCode), uri(uriPathPattern));
+    }
+
+    private KeyValue uri(@Nullable String uriTemplate) {
+        if (uriTemplate != null) {
+            return KeyValue.of(CommonLowCardinalityKeys.URI, extractPath(uriTemplate));
+        }
+        return URI_NONE;
+    }
+
+    private static String extractPath(String uriTemplate) {
+        String path = PATTERN_BEFORE_PATH.matcher(uriTemplate).replaceFirst("");
+        return (path.startsWith("/") ? path : "/" + path);
+    }
+
+    private KeyValue method(@Nullable String methodName) {
+        if (methodName != null) {
+            return KeyValue.of(CommonLowCardinalityKeys.METHOD, methodName);
+        }
+        else {
+            return METHOD_NONE;
+        }
+    }
+
+    private KeyValue status(@Nullable Integer statusCode) {
+        if (statusCode == null) {
+            return STATUS_CLIENT_ERROR;
+        }
+        return KeyValue.of(CommonLowCardinalityKeys.STATUS, String.valueOf(statusCode));
```

Comment:
To make this consistent to the other methods:

```suggestion
        if (statusCode != null) {
            return KeyValue.of(CommonLowCardinalityKeys.STATUS, String.valueOf(statusCode));
        }
        return STATUS_CLIENT_ERROR;
```